examples/type-assertion/type_assertion.bal

```diff
@@ -0,0 +1,28 @@
+import ballerina/io;
+
+type Person record {
+    string name = "";
+    int age = 0;
+};
+
+type Employee record {
+    string name = "";
+    int age = 0;
+    int empNo = 0;
+};
+
+function explainTypeAssertion(anydata emp) returns () {
+    // There is no type casting happens here. Here, emp variable's type is asserted as `Employee` and assigns the same emp
+    // value to employee variable. This is safe type assertion.
```

Comment:
We don't need to say "There is no type casting happens here.", right?

We can just say 
"`emp` is asserted to be of type `Employee` and if successful the value is assigned to variable `employee`"

And, why are we calling this safe type assertion? IMO, the only safe type assertion for non basic types is if we do something like
```ballerina
Employee e = {};
Employee f = <Employee> e;
```